lib/ansible/modules/cloud/amazon/cloudformation.py

```diff
@@ -361,6 +363,34 @@ def stack_operation(cfn, stack_name, operation):
             time.sleep(5)
     return {'failed': True, 'output':'Failed for unknown reasons.'}
 
+
+def get_changeset(module, stack_params, cfn):
+    """Create a change set, get it and delete it."""
+    randomchars = [random.choice(string.ascii_letters + string.digits) for x in range(0, 10)]
+    name = 'ansible-cloudformation-check-' + ''.join(randomchars)
+    try:
+        change_set = cfn.create_change_set(ChangeSetName=name, **stack_params)
+        while True:
+            description = cfn.describe_change_set(ChangeSetName=change_set['Id'])
```

Comment:
Instead of infinite checking, please cap the amount of time to some (can be long) value. 5 or 10 minutes would be *plenty* for CFN to generate a changeset.